### ISSUE TITLE
Indexer cli and backend showing different result

### ISSUE THREAD

**Issue by LouNik1984** [magento/magento2#3675]
Hi,

I installed magento2, changed mode to developer from cli (delploy:mode:show returns Current application mode: developer) and tried to reindex from cli by running magento indexer:reindex .. everything seemed to go fine: I get the list of the indexes and the time it took to rebuild the index. If I check the status from cli with indexer:status, every index is shown as 'ready'. But when I go to backend, all the indexes are shown as 'reindex required' (mode is 'update on save'). Is this a bug? Is there some cli command to run aside the ones I already tried? I checked other issues here and the developer documentation but there seems to be nothing apart the commands I already tried..

**Comment by mazhalai** [comment]
@LouNik1984 Thank you! we have created a ticket MAGETWO-50201 to investigate and fix.

**Comment by katmoon** [comment]
Hi @LouNik1984 

What are indexers statuses in `indexer_state` table in the database?
What are the statuses after you run `bin/magento cron:run; bin/magento cron:run; bin/magento indexer:status` ?
Does Index Management page still display 'Reindex required'?

Thanks,
Kateryna.

**Comment by LouNik1984** [comment]
Hi,
I cannot have cron running on my local environments, I only acitvate it when I upload my site on remote servers, that's why I tried to index manually through cli. After indexing from cli, indexer:status shows me that every index is 'ready' but when I go to backend, the indexes are still red (=invalid) and the message 'reindex required' is still displayed. If I check the table indexer_state, the table also shows the indexes as invalid. So basically it's like the cli commands for reindexing are doing nothing, even if they appear working and show indexes 'as ready'.

**Comment by mslabko** [comment]
Hello @LouNik1984,

I cannot reproduce problem on develop branch. Could you please check it again?
Also we do not cache index status, so result from cli and web must be the same. 

If this is happened again, please check, that you are run bin/magento for correct magento installation (I mean magento is run from the same folder on server, where you run cli commands). E.g. you can move bootstrap.php for break application:
mv app/bootstrap.php app/bootstrap.php.bkp

**Comment by LouNik1984** [comment]
I created a cron on my local environment and after cron has run, indexes were fine. But who wants cron jobs running on a local environment? I hoped that cli commands would work without cron but apparently they do not (in my narrow view 'manual index' is the opposite of 'automatic index').

**Comment by oserediuk** [comment]
@LouNik1984, do you still reproduce this issue? If yes, please, provide your last code revision. Try to update you code and reproduce one more time. I also did not reproduce this issue on my environment

**Comment by LouNik1984** [comment]
Yes, the only way I can have cli index command work is with cron set up.. I haven't the latest code, though, so I'll try a fresh install next week, I'll let you know as soon as I tested it

**Comment by oserediuk** [comment]
@LouNik1984 Ok, I hope it won't reproduce :)

**Comment by LouNik1984** [comment]
I hope that too! I'll write again as soon as I have (hopefully good) news ^^

**Comment by LouNik1984** [comment]
Ok, I've just installed the lastest magento 2 stable version (2.0.6) and some of the issues I found previously are gone (local settings are the same as old installation). I installed M2 from zip (not composer) and:
- install went smoothly from first time (so was my old installation but this time installation was a bit faster)
- first time loading backend and frontend pages right after installation has finished is very slow (so it was previous installation, both times it took more than 10 minutes for pages to be loaded)
- backend and frontend css and images show correctly (in previous installation they didn't load)
- indexing is still weird: cli commands works as expected but only after I set a cron job. If I try to reindex from cli without any cron job set, cli commands show like they're working (they write on the command line the expected results) but then the admin page shows that reindex is still needed (basically it's the same behaviour as previous installation)
- cache refreshing: button in backend for refreshing is available (in previous installation I couldn't find any button, I could only refresh cache from cli); cache refreshing from backend is a bit slow

So there are some improvements with this new version, I hope there will be an admin button for indexing also, I cannot tell clients to refresh indexes from command line ^^ 

ps: I know this issue is only for indexing, but I thought that putting together all I experienced in one place was a bit better, I apologize if I thought wrong.

**Comment by katmoon** [comment]
@LouNik1984 

> indexing is still weird: cli commands works as expected but only after I set a cron job. If I try to reindex from cli without any cron job set, cli commands show like they're working (they write on the command line the expected results) but then the admin page shows that reindex is still needed (basically it's the same behaviour as previous installation)

Cron job that you setup runs the same command as you run manually.
Please note that you need to run the command twice so that indexes are reindexed. See here in the end of the page: http://devdocs.magento.com/guides/v2.0/config-guide/cli/config-cli-subcommands-cron.html

Thanks,
Kateryna

**Comment by LouNik1984** [comment]
"Cron job that you setup runs the same command as you run manually."

My issue here is that I don't want to run the cron on my local environment but if I don't set any cron, no matter how many time I run the indexer command manually, reindex behaves weird: command interface shows successful message as the indexer has worked, but then when I go back to admin page and refresh the page, it tells me that indexing is still required; if then I check the index status with the command line, it says that every index is ready, so basically without cron admin backend and command line are saying different things. If I set a cron job and then reindex from command line, index admin page gets updated as well and shows the indexes as green instead of red, so with cron enabled backend and command line are saying the same thing.

**Comment by dimple-vasoya** [comment]
Hello,

I have upgraded my Magento version from 2.1.0 to 2.1.7
Earlier the reindex time for Product Categories was 00:03:08 and now it has increased to 01:12:00
Can anyone help me understand the reason behind this?

**Comment by korostii** [comment]
@dimple-ambab,
Hi. Please create a new separate issue instead of flooding existing ones. That'll increase the chances of getting a faster response.

**Comment by mslabko** [comment]
@dimple-ambab , Could you please provide us a dump of your database for 2.1.0 so we can investigate it. For our medium profile we cannot find the issue with slow indexation mslabko@magento.com

**Comment by magento-engcom-team** [comment]
@LouNik1984, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9

**Comment by magento-engcom-team** [comment]
@LouNik1984, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9
